### PR TITLE
Allow PayloadPlace to be compiled out, factor same

### PR DESCRIPTION
**Pull request by peterbarker**
This work sponsored by FreeSpace Operations.

This saves ~1600 bytes on minimized boards, which lose this feature, and a few bytes on other boards.

There are two new defines, one which takes out the "payload place flight behaviour" (`AC_PAYLOAD_PLACE_ENABLED`), and one which takes out the payload place mission items (`AP_MISSION_NAV_PAYLOAD_PLACE_ENABLED`).  You won't compile if the mission item is enabled but the flight behaviour is not.

Reason for the two defines is that this "payload place" flight behaviour is slated to be enabled for "ship operations", so you can use the descent and payload release behaviour outside of auto mode.

This also cleans up the safe-ground-handling code to make it more consistent with other similar blocks.

### CODE REVIEW

**Comment by peterbarker** [comment]
```
Board               AP_Periph  blimp  bootloader  copter  heli   iofirmware  plane  rover  sub
Durandal                       *      *           -72     -72                *      *      *
HerePro             *                                                                      
Hitec-Airspeed      *                                                                      
KakuteH7-bdshot                *      *           -88     -96                *      *      *
MatekF405                      *      *           -112    -88                *      *      *
Pixhawk1-1M-bdshot             -56                -1720   -1720              -56    -56    -56
f103-QiotekPeriph   *                                                                      
f303-Universal      *                                                                      
iomcu                                                            *                         
revo-mini                      -56    *           -1728   -1720              -56    -48    -56
skyviper-v2450                                    -96                                      
```

File: libraries/AP_Mission/AP_Mission.cpp [hunk 1]
```diff
@@ -1753,9 +1755,11 @@ bool AP_Mission::mission_cmd_to_mavlink_int(const AP_Mission::Mission_Command& c
         packet.param3 = cmd.content.do_engine_control.height_delay_cm*0.01f;
         break;
 
+#if AP_MISSION_NAV_PAYLOAD_PLACE_ENABLED
     case MAV_CMD_NAV_PAYLOAD_PLACE:
         packet.param1 = cmd.p1/100.0f; // copy max-descend parameter (cm->m)
```
Comment (amilcarlucas) [inline libraries/AP_Mission/AP_Mission.cpp:1760]: Driveby, but can you use a *0.01f here instead?

Comment (peterbarker) [reply libraries/AP_Mission/AP_Mission.cpp:1760]: I'd rather not complicate the PR... especially as this is not exactly a "hot" path :-)

I'll be working more in this area, so will slip that change in in a future PR :-)

**Comment by IamPete1** [comment]
The name spacing and defines all look good to me. There should be no change in behaviour there. 

There are some behavior changes here I think just commits, [00082d187862538f7e2447cdef1bef429d400e2d ](https://github.com/ArduPilot/ardupilot/pull/25310/commits/00082d187862538f7e2447cdef1bef429d400e2d) and [8c3935a222a50cdfbaaa31787767f05cb864cea5](https://github.com/ArduPilot/ardupilot/pull/25310/commits/8c3935a222a50cdfbaaa31787767f05cb864cea5)

**Review by tridge — CHANGES_REQUESTED** [review]
add to extract_features and build_options

**Review by IamPete1 — APPROVED** [review]
Code looks good, the functional changes are from/have been reviewed by @lthall

**Comment by peterbarker** [comment]
> add to extract_features and build_options

Done
